src/client.js

```diff
@@ -136,14 +136,29 @@ export default class StitchClient {
    */
   authenticate(providerType, options = {}) {
     // reuse existing auth if present
-    return this.auth.getAccessToken().then(accessToken => {
+    return Promise.all(
+      [
+        this.auth.getAccessToken(),
+        this.auth.getLoggedInProviderType()
+      ]
+    ).then(([accessToken, loggedInProviderType]) => {
       if (accessToken) {
-        return this.auth.authedId();
+        // TODO: `anon` should be enumerated once providers are maintain
+        // an interface
+        if (providerType === 'anon' && loggedInProviderType === 'anon') {
+          return false; // is authenticated, skip log in
+        }
+
+        return this.logout(); // will not be authenticated, continue log in
       }
 
-      return this.auth.provider(providerType).authenticate(options)
-        .then(() => this.auth.authedId());
-    });
+      return true; // is not authenticated, continue log in
+    }).then((isNotAuthed) => {
+      if (isNotAuthed) {
+        return this.auth.provider(providerType).authenticate(options);
+      }
+      return;
+    }).then(() => this.auth.authedId());
```

Comment:
[q] Can just pass `this.auth.authedId` instead of having an arrow function, right?

Reply:
Probably. Will have to try.

Reply:
Apparently not, unless I'm doing it wrong.

Reply:
nah makes sense, the this here is the client